packages/next/server/config.ts

```diff
@@ -77,13 +95,31 @@ function assignDefaults(userConfig: { [key: string]: any }) {
         return currentConfig
       }
 
-      if (
-        key === 'experimental' &&
-        value !== defaultConfig[key] &&
-        typeof value === 'object' &&
-        Object.keys(value).length > 0
-      ) {
-        experimentalWarning(configFileName, Object.keys(value))
+      if (key === 'experimental' && typeof value === 'object') {
+        const notExistedExperimentalFeatures: string[] = []
+        const enabledExperimentalFeatures: (keyof ExperimentalConfig)[] = []
+
+        for (const featureName of Object.keys(
+          value
+        ) as (keyof ExperimentalConfig)[]) {
+          if (!(featureName in defaultConfig.experimental!)) {
+            notExistedExperimentalFeatures.push(featureName)
+          } else if (
+            value[featureName] !== defaultConfig.experimental![featureName]
```

Comment:
Lets hoist `defaultConfig.experimental` up and check for existence on line 98 so we can remove the `!` TS assertion

Reply:
`defaultConfig.experimental` is predefined in `config-shard.ts` and will never be undefined. But I will add a type guard just to avoid the non-null assertion.